### ISSUE TITLE
enricher, loader, and mutator have defective health endpoints

### ISSUE THREAD

**Issue by mabushey** [snowplow/snowplow#4514]
This issue does not appear to apply to collector and repeator. In a GKE k8s cluster (v1.20.8-gke.2100), enricher, loader, and mutator all give `Liveness probe failed: HTTP probe failed with statuscode: 500` within a minute or two.
```
          livenessProbe:
            httpGet:
              path: /health
              port: 8080
              scheme: HTTP
            initialDelaySeconds: 30
            timeoutSeconds: 10
            failureThreshold: 3
            successThreshold: 1
```

Is /health on 8080 the wrong endpoint or are the services just locking up every couple of minutes?

- image: snowplow/snowplow-enrich-pubsub:2.0.2
- image: snowplow/snowplow-bigquery-streamloader:0.6.4
- image: snowplow/snowplow-bigquery-mutator:0.6.4

**Comment by mabushey** [comment]
I guess they literally don't have health endpoints. Unfortunately I'm running on GKE which has no clue when a preemptable node goes away. The containers don't even have `ps` installed, so I'm just using this:
```
          livenessProbe:
            exec:
              command:
              - /bin/sh
              - -c
              - /bin/hostname
            initialDelaySeconds: 60
            timeoutSeconds: 10
            failureThreshold: 3
            successThreshold: 1
```

**Comment by jrluis** [comment]
Hi @mabushey,

Indeed, there are no health endpoints on those services.

Have you noticed any wrong/strange behavior if you remove the liveness probe?

There shouldn't exist any issue when a node goes down, even when using preemptible nodes. 

The services are designed to use pubsub in a way that won't loose any data, even if the services crash or the nodes they are running also crash (or in the case of the preemptible nodes, they are shutdown).

It is possible to create gcp alarms so there is visibility on what is happening with the containers e.g. when a container gets rebooted or errors happen.

Regards

**Comment by mabushey** [comment]
HI @jrluis ,

Thank you for responding. :) Yes, Those three services were dead for 20 minutes yesterday after three preemptable nodes were replaced. I had to manually terminate the pods to get new ones running. Today (with the /bin/hostname probe) they are running fine, no restarts or issues so it may take some time to find out. They might not currently be running on the preemptable nodes. 

Thanks for the info on pubsub data reliability, that answered a big question/concern we had.

Have a wonderful day. :)
